pkg/koordlet/statesinformer/impl/states_pod_resources.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package impl
+
+import (
+	"context"
+	"net"
+	"os"
+	"path/filepath"
+
+	"google.golang.org/grpc"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+	"github.com/koordinator-sh/koordinator/pkg/features"
+)
+
+const (
+	podResourcesInformerName PluginName = "podResourcesInformer"
+)
+
+var (
+	_ podresourcesapi.PodResourcesListerServer = &podResourcesServer{}
+)
+
+type podResourcesInformer struct {
+	config         *Config
+	nodeInformer   *nodeInformer
+	resourceServer podresourcesapi.PodResourcesListerServer
+}
+
+func newPodResourcesInformer() *podResourcesInformer {
+	return &podResourcesInformer{}
+}
+
+func (s *podResourcesInformer) Setup(ctx *PluginOption, states *PluginState) {
+	s.config = ctx.config
+
+	nodeInformerIf := states.informerPlugins[nodeInformerName]
+	nodeInformer, ok := nodeInformerIf.(*nodeInformer)
+	if !ok {
+		klog.Fatalf("node informer format error")
+	}
+	s.nodeInformer = nodeInformer
+}
+
+func (s *podResourcesInformer) Start(stopCh <-chan struct{}) {
```

Comment:
please consider graceful exiting
perhaps we need to pass the channel to close the client and stop the server